packages/ui5-application-inquirer/src/prompts/prompts.ts

```diff
@@ -0,0 +1,473 @@
+        [promptNames.ui5Version]: getUI5VersionPrompt(ui5Versions, promptOptions?.ui5Version),
+        [promptNames.addDeployConfig]: getAddDeployConfigPrompt(
+            targetDir,
+            promptOptions?.addDeployConfig,
+            isCapProject
+        ),
+        [promptNames.addFlpConfig]: getAddFlpConfigPrompt(promptOptions?.addFlpConfig),
+        [promptNames.showAdvanced]: getShowAdvancedPrompt(),
+        [promptNames.ui5Theme]: getUI5ThemePrompt(),
+        [promptNames.enableEslint]: getEnableEsLintPrompt(),
+        [promptNames.enableCodeAssist]: getEnableCodeAssistPrompt(),
+        [promptNames.skipAnnotations]: getSkipAnnotationsPrompt(),
+        [promptNames.enableNPMWorkspaces]: getEnableNPMWorkspacesPrompt(capCdsInfo),
+        [promptNames.enableTypeScript]: getEnableTypeScriptPrompt(capCdsInfo)
+    };
+
+    // Hide not applicable prompts based on passed options or if this is a CAP project
+    let questions: UI5ApplicationQuestion[] = hidePrompts(keyedPrompts, promptOptions, isCapProject);
+
+    // Add an additonal condition to 'advanced' prompts so they can be shown/hidden at runtime
+    applyAdvancedOption(questions, promptOptions);
+
+    // Apply extended `validate`, `additionalMessages` or override `default` prompt properties
+    if (promptOptions) {
+        questions = extendWithOptions(questions, promptOptions);
+    }
+    return questions;
+}
+
+/**
+ * Get the `enableTypeScript` prompt.
+ *
+ * @param capCdsInfo CDS UI5 plugin information
+ * @returns The `enableTypeScript` prompt
+ */
+function getEnableTypeScriptPrompt(capCdsInfo?: CdsUi5PluginInfo): UI5ApplicationQuestion {
+    return {
+        when: (answers): boolean => {
+            if (capCdsInfo) {
+                return capCdsInfo.isCdsUi5PluginEnabled || !!answers?.enableNPMWorkspaces;
+            }
+            return true;
+        },
+        type: 'confirm',
+        name: promptNames.enableTypeScript,
+        message: t('prompts.appEnableTypeScriptMessage'),
+        default: false,
+        guiOptions: {
+            breadcrumb: true
+        }
+    } as ConfirmQuestion<UI5ApplicationAnswers>;
+}
+
+/**
+ * Get the `enableNPMWorkspaces` prompt.
+ *
+ * @param capCdsInfo CDS UI5 plugin information
+ * @returns The `enableNPMWorkspaces` prompt
+ */
+function getEnableNPMWorkspacesPrompt(capCdsInfo?: CdsUi5PluginInfo): UI5ApplicationQuestion {
+    return {
+        when: () => {
+            if (capCdsInfo) {
+                return capCdsInfo.hasMinCdsVersion && !capCdsInfo.hasCdsUi5Plugin;
+            }
+            return false;
+        },
+        type: 'confirm',
+        name: promptNames.enableNPMWorkspaces,
+        message: t('prompts.appEnableNpmWorkspacesMessage'),
+        default: false,
+        guiOptions: {
+            breadcrumb: t('prompts.appEnableNpmWorkspacesBreadcrumb')
+        }
+    } as ConfirmQuestion<UI5ApplicationAnswers>;
+}
+
+/**
+ * Get the `skipAnnotations` prompt.
```

Comment:
If I am not mistaken then this prompt is only valid for CAP, therefore, could you please enhance the comment that it easier to understand what this prompt is about.

Reply:
Done